src/ORM/DataObject.php

```diff
@@ -2834,55 +2834,52 @@ public function setField($fieldName, $val)
 
         // Situation 1: Passing an DBField
         if ($val instanceof DBField) {
-            $val->setName($fieldName);
-            $val->saveInto($this);
```

Comment:
It looks like a lot has changed in this file though it hasn't - previously it was:
```php
if ($val instanceof DBField) {
  // handle DBField
} else {
  // handle non-DBField $val
}
```

Now it is

```php
if ($val instanceof DBField) {
  $val = $val->getValue()
}
// handle non-DBField $val
```

The reason this was done was was to fix a dynamic property deprecation warning caused by SecurityTest::testExternalBackUrlRedirectionDisallowed() where at attempt was made to do a magic-method __set() using a DBField as a value

```
ChangePasswordHandler::doChangePassword()
  $member->AutoLoginExpired = DBDatetime::create()->now(); << value is set as a DBField
ViewableData::__set() <<< first call to __set()
DataObject::setField()
DBField::saveInto()
  $dataObject->$fieldName = $this->value; <<< can't re-call __set(), dynamic property deprecation warning
```